frontend/Store.js

```diff
@@ -419,12 +459,33 @@ class Store extends EventEmitter {
     });
   }
 
-  changeBananaSlug(state: Object) {
+  changeBananaSlug(state: ControlState) {
     this.bananaslugState = state;
     this.emit('bananaslugchange');
+    invariant(state.toJS);
```

Comment:
would be cool to remove the invariants and maybe use https://flowtype.org/docs/union-intersection-types.html of `import type {Record} from 'immutable'` and `{enabled: boolean}`